src/component/ResponsiveContainer.js

```diff
@@ -7,56 +7,70 @@ import pureRender from '../util/PureRender';
 import { getPercentValue, isPercent } from '../util/DataUtils';
 import { warn } from '../util/LogUtils';
 
-const render = ({ width, height, container, children }) => {
+const render = ({ aspect, width, height, minWidth, minHeight, container, children }) => {
+
   warn(isPercent(width) || isPercent(height),
-    `The width(%s) and height(%s) are both fixed number,
-     maybe you don't need to use ResponsiveContainer.`,
+    `The width(%s) and height(%s) are both fixed numbers,
+     maybe you don't need to use a ResponsiveContainer.`,
     width, height
   );
 
-  const calculatedWidth = getPercentValue(width, container.width);
-  const calculatedHeight = getPercentValue(height, container.height);
+  warn(!aspect || aspect > 0,
+       'The aspect(%s) must be greater than zero.',
+       aspect);
 
-  warn(calculatedWidth > 0 && calculatedHeight > 0,
-    `The width(%s) and height(%s) of chart should be greater than 0,
-    please check the style of container, or the props width(%s) and height(%s).`,
-     calculatedWidth, calculatedHeight, width, height
-  );
+  let calculatedWidth = getPercentValue(width, container.width);
```

Comment:
Because we set the width, height, minHeight, minWidth in [L50](https://github.com/recharts/recharts/pull/192/commits/aec73d638ae9fa14c5e52e543934eb8f1b599d21#diff-64e08d9691c7f3156ef1dfd43b7c94e5R50), `container.width` should be the width we need to render charts.

```
let calculatedWidth = container.width;
let calculatedHeight = container.height;
```